gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ProcedureRequestMapperTest.java

```diff
@@ -122,14 +133,85 @@ public void mapProcedureRequestWithDegradedPlan() {
+    }
+
+    @Test
+    public void mapProcedureRequestMetaSecurityWithNoPatWhenConfidentialityCodeIsPresentForPlanStatementAndEhrComposition() {
+        var ehrExtract = unmarshallCodeElement("no_optional_data_example.xml");
+
+        var planStatement = getPlanStatement(ehrExtract);
+        planStatement.setConfidentialityCode(NOPAT_CV);
+        var ehrComposition = getEhrComposition(ehrExtract);
+        ehrComposition.setConfidentialityCode(NOPAT_CV);
+
+        when(confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+            META_PROFILE,
+            ehrComposition.getConfidentialityCode(),
+            planStatement.getConfidentialityCode()
+        )).thenReturn(META);
+
+        ProcedureRequest procedureRequest = procedureRequestMapper.mapToProcedureRequest(ehrComposition,
+                                                                                         planStatement, SUBJECT, ENCOUNTERS, PRACTISE_CODE);
+
+        assertNotNull(procedureRequest);
+        assertMetaSecurityIsPresent(procedureRequest.getMeta());
+    }
+
+    @Test
+    public void mapProcedureRequestMetaSecurityWithNoPatWhenConfidentialityCodeIsPresentOnlyForEhrComposition() {
+        var ehrExtract = unmarshallCodeElement("no_optional_data_example.xml");
+        var planStatement = getPlanStatement(ehrExtract);
+        var ehrComposition = getEhrComposition(ehrExtract);
+        ehrComposition.setConfidentialityCode(NOPAT_CV);
+
+        when(confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+            META_PROFILE,
+            ehrComposition.getConfidentialityCode(),
+            Optional.empty()
+        )).thenReturn(META);
+
+        ProcedureRequest procedureRequest = procedureRequestMapper.mapToProcedureRequest(ehrComposition,
+                                                                                         planStatement, SUBJECT, ENCOUNTERS, PRACTISE_CODE);
+
+        assertNotNull(procedureRequest);
+        assertMetaSecurityIsPresent(procedureRequest.getMeta());
+    }
+
+    @Test
+    public void mapProcedureRequestMetaSecurityWithNoScrubWhenConfidentialityCodeIsNotPresent() {
+        var ehrExtract = unmarshallCodeElement("no_optional_data_example.xml");
+        var planStatement = getPlanStatement(ehrExtract);
+        var ehrComposition = getEhrComposition(ehrExtract);
+        ehrComposition.setConfidentialityCode(ALTERNATIVE_CV);
+
+        when(confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+            META_PROFILE,
+            ehrComposition.getConfidentialityCode(),
+            Optional.empty()
+        )).thenReturn(ALTERNATIVE_META);
+
+        ProcedureRequest procedureRequest = procedureRequestMapper.mapToProcedureRequest(ehrComposition,
+                                                                                         planStatement, SUBJECT, ENCOUNTERS, PRACTISE_CODE);
```

Comment:
Formatting looks a bit funny here too, this may be the preferred formatting though so I'm unsure if this can be done:

```suggestion
        ProcedureRequest procedureRequest = procedureRequestMapper.mapToProcedureRequest(
            ehrComposition,
            planStatement, 
            SUBJECT, 
            ENCOUNTERS, 
            PRACTISE_CODE
        );
```

Reply:
done